.github/workflows/automerge.yml

```diff
@@ -22,16 +22,110 @@ env:
     # https://github.com/JuliaRegistries/RegistryCI.jl/blob/ee1d7cdb165202f4f3929a122c3188fbdd7afc16/src/AutoMerge/ciservice.jl#L53-L67
     NEED_MERGE_TOKEN: ${{ github.ref == 'refs/heads/master' && (github.event_name == 'schedule' || github.event_name == 'workflow_dispatch') }}
 
+# We have three jobs:
+# 1. AutoMerge-regular:
+#    This includes cron runs, workflow_dispatch runs, and all PR runs except PR labels.
+# 2. AutoMerge-pr-label
+#    This only includes PR label runs.
+#    It is important to keep this job separate.
+#    If we don't keep this job separate, then any label action will override the main/regular
+#    AutoMerge job with a "skipped" AutoMerge job, thus making it very difficult for PR authors
+#    to view the CI logs (which they need to do if they e.g. want to figure out why `import MyPackageName`
+#    failed during the AutoMerge run.)
+# 3. AutoMerge-stopwatch
 jobs:
-  AutoMerge:
+  AutoMerge-regular:
```

Comment:
could we reduce duplication by setting the job name dynamically based on the trigger? https://github.blog/changelog/2022-09-26-github-actions-dynamic-names-for-workflow-runs/

Reply:
Alright, I tried setting the job name dynamically: https://github.com/JuliaRegistries/General/pull/130157

Unfortunately, as can be seen in that PR, GitHub will still overwrite the jobs with each other. The synchronize/close/reopen job will override the label job, and the label job will subsequently overwrite the synchronize/close/reopen job (with a "skipped" status).

Reply:
Ah ok, thanks for looking into it. That’s too bad, let’s go with this approach then